packages/babel-parser/src/tokenizer/state.ts

```diff
@@ -221,7 +229,9 @@ export type LookaheadState = {
   lastTokEndLoc: Position;
   curLine: number;
   lineStart: number;
-  curPosition: () => Position;
+  curPosition: State["curPosition"];
+  sourceToOffsetPos: State["sourceToOffsetPos"];
+  offsetToSourcePos: State["offsetToSourcePos"];
```

Comment:
The `lookaheadState` should be kept minimal for performance. I guess we didn't call `sourceToOffsetPos` or `offsetToSourcePos` on a lookahead state, if that is true, then we don't have to include them in the lookahead state.

Reply:
@JLHwung fyi without this the tests are failing and so it seems necessary with the changes in the PR

Reply:
Can we move those functions from State to the parser itself maybe?

Reply:
If we inline `sourceToOffsetPos` in `curPosition`, the test passes.

Reply:
I pushed a commit to move them.